e2e/bridge_test.go

```diff
@@ -0,0 +1,264 @@
+		t,
+		IBFTMinNodes,
+		fmt.Sprintf("%sdest", IBFTDirPrefix),
+		func(i int, config *framework.TestServerConfig) {
+			config.SetSeal(true)
+			config.Premine(senderAddr, framework.EthToWei(10))
+			config.SetUseBridge(true)
+			config.SetBridgeRootChainURL(sourceIBFT.GetServer(0).WSJSONRPCAddr())
+			config.SetBridgeRootChainContract(syncerContractAddr.String())
+			config.SetBridgeRootChainConfirmations(5)
+		})
+
+	startDestIBFTCtx, startDestIBFTCancel := context.WithTimeout(context.Background(), time.Minute)
+	defer startDestIBFTCancel()
+	destIBFT.StartServers(startDestIBFTCtx)
+
+	// deploy state receiver contract
+	deployStateReceiverCtx, deployStateReceiverCancel := context.WithTimeout(
+		context.Background(),
+		time.Minute,
+	)
+	defer deployStateReceiverCancel()
+
+	reciverContractAddr, err := destIBFT.GetServer(0).DeployContract(
+		deployStateReceiverCtx,
+		StateReceiverByteCode,
+		senderKey,
+	)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// syncData is sent to StateSync contract and forwarded to StateReceiver
+	syncData := "hello"
+
+	// Send state sync contract
+	to := types.BytesToAddress(syncerContractAddr[:])
+	input, err := StateSyncerABI.Methods["stateSync"].Encode(map[string]interface{}{
+		"contractAddress": reciverContractAddr.String(),
+		"data":            []byte(syncData),
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sendStateSyncCtx, sendStateSyncCancel := context.WithTimeout(context.Background(), time.Minute)
+	defer sendStateSyncCancel()
+
+	_, err = sourceIBFT.GetServer(0).SendRawTx(sendStateSyncCtx, &framework.PreparedTransaction{
+		Gas:      framework.DefaultGasLimit,
+		GasPrice: big.NewInt(framework.DefaultGasPrice),
+		To:       &to,
+		From:     senderAddr,
+		Input:    input,
+	}, senderKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = waitUntilStateReachesToContract(
+		context.Background(),
+		destIBFT.GetServer(0),
+		senderAddr,
+		reciverContractAddr,
+		syncData,
+	)
+	assert.NoError(t, err)
+}
+
+// waitUntilStateReachesToContract repeats to get state in StateReceiver contract until expected state is set
+func waitUntilStateReachesToContract(
+	ctx context.Context,
+	srv *framework.TestServer,
+	senderAddress types.Address,
+	contractAddress web3.Address,
+	expectedState string,
+) error {
+	_, err := tests.RetryUntilTimeout(ctx, func() (interface{}, bool) {
+		response, err := srv.JSONRPC().Eth().Call(
+			&web3.CallMsg{
+				From:     web3.Address(senderAddress),
+				To:       &contractAddress,
+				Data:     StateRecieverABI.Methods["getState"].ID(),
+				GasPrice: 100000000,
+				Value:    big.NewInt(0),
+			},
+			web3.Latest,
+		)
+		if err != nil {
+			return err, true
+		}
+
+		byteData, err := types.ParseBytes(&response)
+		if err != nil {
+			return err, true
+		}
+
+		res, err := StateRecieverABI.Methods["getState"].Outputs.Decode(byteData)
+		if err != nil {
+			return err, true
+		}
+
+		resMap, ok := res.(map[string]interface{})
+		if !ok {
+			return nil, false
+		}
```

Comment:
Nitpick, this should be a separate method

Reply:
Added a function to get state from StateReceiver at 1755956c356860b0b10fd95865c122196138b3b3